### DISCUSSION TITLE
Can apply migrations but authentication fails during runtime

### DISCUSSION THREAD

**Discussion by SammyWhamy** [prisma/prisma#13762] [Q&A]
I just started hosting a PostgreSQL database on an external server instead of locally, so that I can get ready for production. I spent a long time figuring out how to connect in the first place, but that seems to be working now!

I can successfully use a tool like DataGrip to connect to the database and view tables and data. 
Running `npx prisma migrate dev` also runs successfully, and I can confirm that the tables have been created in the external database.
However, when I now start my application and try to do any action on the database, I get this:
```
error - Error:
Invalid `prisma.token.findMany()` invocation:


  Authentication failed against database server at `xxx.xxx.xxx.xxx`, the provided database credentials for `user` are not valid.

Please make sure to provide valid database credentials for the database server at `xxx.xxx.xxx.xxx`.
```

I have two env files, `.env` which only contains `DATABASE_URL` and `.env.local` which contains other secrets used in my next.js application.

My schema looks like this:
```sql
# Model definitions

generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}
```

I cannot wrap my head around how the migrations through prisma are working perfectly fine, while it's not working during runtime. Any help is appreciated!

Update: It seems that the exact same connection string functions in prod when deployed to vercel, but not from localhost

**Comment by janpio** [comment]
Prisma by default only loads `.env` into the environment, so when you ran `migrate dev` you either did that explicitly for `.env.local` as well, or you accidentally had the `DATABASE_URL` in your environment from some other source (like playing around before).

**Reply by SammyWhamy** [reply]
The `DATABASE_URL` is in the `.env` file that prisma suggests you to use, not in the `.env.local` one, prisma doesn't need anything from `.env.local`

Edit: I see I mixed them up in my original question, will edit that!

**Reply by janpio** [reply]
Then I see no logical reason why something that works in `migrate dev` should not work when using Prisma Client. Especially as it says it is connecting to the correct database hostname. Sorry.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
